octodns/provider/cloudflare.py

```diff
@@ -18,7 +19,7 @@ class CloudflareError(Exception):
     def __init__(self, data):
         try:
             message = data['errors'][0]['message']
-        except (IndexError, KeyError):
+        except (IndexError, KeyError, TypeError):
```

Comment:
What resulted in needing the `TypeError` here?

Reply:
I added it to catch responses that were not valid json. Example returning `boo` instead of `{}`. When I copied the tests from NS1, it seemed like a good case to handle. I can see how it is only semi related and I can remove it this PR if that makes sense.

Reply:
Seems fine, just wanted to make sure it was intentional.